### PR TITLE
Feat/td 28533

### PR DESCRIPTION
**Pull request by sheyanjie-qq**

### CODE REVIEW

File: cmd/command.go [hunk 1]
```diff
@@ -0,0 +1,444 @@
...
+		if _, err := cmd.conn.Exec(ctx, "DROP STABLE IF EXISTS "+stable); err != nil {
+			logger.Errorf("drop stable %s, error: %s", stable, err)
+		}
+	}
+	logger.Info("## drop old taosd metric stables success!!")
+}
+
+func (cmd *Command) TransferDataToDest(data *db.Data, dstTable string, tagNum int) error {
+
+	var buf bytes.Buffer
```
Comment (huskar-t) [inline cmd/command.go:218]: use reference

File: cmd/command.go [hunk 2]
```diff
@@ -0,0 +1,444 @@
...
+			}
+		}
+		buf.WriteString(" ")
+
+		// write metrics
+		for j := tagNum; j < len(row)-1; j++ {
+
+			switch v := row[j].(type) {
+			case int:
+				buf.WriteString(fmt.Sprintf("%s=%ff64", data.Head[j], float64(v)))
```
Comment (huskar-t) [inline cmd/command.go:258]: fmt.Sprintf will lose float precision

File: cmd/command.go [hunk 3]
```diff
@@ -0,0 +1,444 @@
...
+		// write timestamp
+		buf.WriteString(" ")
+		buf.WriteString(fmt.Sprintf("%v", row[len(row)-1].(time.Time).UnixMilli()))
+		buf.WriteString("\n")
+
+		if buf.Len() >= MAX_SQL_LEN {
+			if logger.Logger.IsLevelEnabled(logrus.TraceLevel) {
+				logger.Tracef("## buf: %v", buf.String())
+			}
+			cmd.lineWriteBody(buf)
```
Comment (huskar-t) [inline cmd/command.go:285]: Unhandled error

File: cmd/command.go [hunk 4]
```diff
@@ -0,0 +1,444 @@
...
+			current.UnixMilli(), current.Add(time.Duration(delta)).UnixMilli())
+		logger.Tracef("## query sql: %s", querySql)
+		data, err := cmd.conn.Query(ctx, querySql)
+		if err != nil {
+			logger.Errorf("query cluster_info error: %s", err)
+			return err
+		}
+
+		// transfer data to new table, only this table need use insert statement
+		cmd.TransferDataToDest(data, dstTable, tagNum)
```
Comment (huskar-t) [inline cmd/command.go:407]: Unhandled error

File: cmd/command.go [hunk 5]
```diff
@@ -0,0 +1,444 @@
...
+		return
+	}
+
+	if len(conf.Drop) > 0 && conf.Drop != "old_taosd_metric_stables" {
+		logger.Errorf("drop only support old_taosd_metric_stables")
+		return
+	}
+
+	if conf.Drop == "old_taosd_metric_stables" {
+		cmd.ProcessDrop(conf)
```
Comment (huskar-t) [inline cmd/command.go:89]: If the transfer fails, the drop will still be executed

Comment (sheyanjie-qq) [reply cmd/command.go:94]: add check

**Review by huskar-t — APPROVED** [review]

**Pull request merged**